js/lib/baseDomain.js

```diff
@@ -9,15 +9,32 @@ const LRUCache = require('lru_cache/core').LRUCache
 
 let cachedBaseDomain = new LRUCache(50)
 
+const checkASCII = function (str) {
+  if (typeof str !== 'string') {
+    return false
+  }
+  for (let i = 0; i < str.length; i++) {
+    if (str.charCodeAt(i) > 127) {
+      return false
+    }
+  }
+  return true
+}
+
 /**
  * Returns base domain for specified host based on Public Suffix List.
  * @param {string} hostname The name of the host to get the base domain for
  */
 
 module.exports.getBaseDomain = function (hostname) {
   // decode punycode if exists
-  if (hostname.indexOf('xn--') >= 0) {
-    hostname = punycode.toUnicode(hostname)
+  if (hostname.indexOf('xn--') >= 0 &&
+    checkASCII(hostname)) {
+    try {
```

Comment:
++ on the try/catch! You'd be surprised that some domains look legit punycoded but actually fail the decoding process.  Not having try/catch has bitten me multiple times in the past :smile: